src/Cli/dotnet/commands/dotnet-test/SolutionAndProjectUtility.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.RegularExpressions;
+using Microsoft.Build.Construction;
```

Comment:
Please don't use this namespace - instead use the new SLNX file support in Microsoft.VisualStudio.SolutionPersistence. This gives you support for both sln and slnx files transparently. cc @edvilme who did most of the work on the .NET CLI's support for SLNX.

No teams/commands should be using the solution parsing directly in MSBuild anymore in the CLI.

Reply:
@baronfel Hello, the old library (`SlnFile`) used to provide a method for obtaining paths of sln files. Closest alternative right now is https://github.com/dotnet/sdk/blob/a93a592ce99414d94dafbd0ef0a6f51b6fbcdaf4/src/Cli/dotnet/commands/dotnet-sln/SlnCommandParser.cs#L51 in `SlnCommandParser`. It seemed simple enough to make it into its own utility class, but I can look into it :)